### PR TITLE
Did Somebody Say Arisaka?

### PR DESCRIPTION
**Pull request by DoctorBoomstick**
#### Summary
Content " Adds most kinds of Arisaka rifles to the game, cartridges, and item spawns."
#### Purpose of change
What country has some cool national culture that survives in CDDA through body pillows and raw fish meals? Japan.
What’s the one thing we’re known for having, apart from body pillows? Guns.
What’s the one thing that makes people do bigger eye rolls the more that get added? That’s right, guns.
Clearly, this means we’re in desperate need of Japanese guns. Suck it, people who think we’ve got too many guns. I agree with you, but the addiction is strong and unyielding.
On a more straight-forward manner of speech:
- I realised how much I like Arisakas.
- I realised how much I like CDDA.
- I realised how much more I’d like those two if they were brought together.
- I smashed them together.

#### Describe the solution
I’ve added the 6.5x50mm and 7.7x58mm Arisaka rounds, plus a modestly-sized buttload of Arisaka rifles themselves.
#### Describe alternatives you've considered
- Adding the Type 2 paratrooper rifle, but I couldn’t figure out a way to simulate its takedown properties in a way that wasn’t stupid.
- Adding the Type I 6.5mm rifle, but I’ve probably inundated the guns folder with enough Arisakas that fire rounds that probably nobody will ever find.
- I considered adding the Nambu Type 96 and 99 light machine guns, plus the Type 11 and Type 92 Heavy machine gun, but I thought that they’d be rare enough to not bother adding them, and my eyes were glazing over from staring at Arisakas for the last 48 hours.
- Adding the Type 14 pistol, Type 100 submachine gun, and other non-rifle calibre weapons, but I didn’t feel like including the 8mm Nambu. I don’t even think it’s made anymore.
 
#### Testing
None, because I wanted to push this PR out before I went on vacation for the next amount of mortal time.

### CODE REVIEW

File: data/json/items/gun/303.json [hunk 1]
```diff
@@ -52,7 +52,7 @@
     "looks_like": "modular_ar15",
     "type": "GUN",
     "name": { "str": "updated Enfield rifle" },
-    "description": "An evolution of Britten’s First World War battle rifle, this bolt-action firearm uses the same .303-caliber cartridge as its forerunner.  Much beloved across the globe by collectors, rifles like this served as England’s infantry rifle well into the 1950s, with a specialized sniper variant remaining in service until the early nineties.",
+    "description": "An evolution of Brittain’s First World War battle rifle, this bolt-action firearm uses the same .303-caliber cartridge as its forerunner.  Much beloved across the globe by collectors, rifles like this served as England’s infantry rifle well into the 1950s, with a specialized sniper variant remaining in service until the early nineties.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/303.json:55]:
```suggestion
    "description": "An evolution of Britain’s First World War battle rifle, this bolt-action firearm uses the same .303-caliber cartridge as its forerunner.  Much beloved across the globe by collectors, rifles like this served as England’s infantry rifle well into the 1950s, with a specialized sniper variant remaining in service until the early nineties.",
```

File: data/json/items/gun/77mm_jap.json [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    "description": "A stout and rugged battle rifle that traces its origins to the hands of imperial Japanese forces and saw extensive use in the Pacific theatre of the Second World War, this bolt-action firearm shoots the 7.7x58mm Arisaka cartridge.  Powerful, dependable, and elegantly built, Japan’s 20th-century infantry rifle may yet find service guarding the lives of her former enemies.",
+    "variant_type": "gun",
+    "variants": [
+      {
+        "id": "type99",
+        "name": { "str_sp": "Arisaka Type 99" },
+        "description": "The standard service arm of the imperial Japanese army during the Second World War, chambered for the powerful 7.7x58mm cartridge and built to replace the weaker Type 38, the Type 99 is a sturdy and formidable bolt-action rifle with a Mouser-inspired action renowned for its strength.  Though the rifle was historically sneered at in the US due to extensive anti-Japanese propaganda and the lower quality of later-production guns, the weapon is, by most metrics, a finely produced and effective shoulder arm.  This particular Arisaka is an early-war example with anti-aircraft sights and an intact 6-pettle chrysanthemum seal.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/77mm_jap.json:14]:
```suggestion
        "description": "The standard service arm of the imperial Japanese army during the Second World War, chambered for the powerful 7.7x58mm cartridge and built to replace the weaker Type 38, the Type 99 is a sturdy and formidable bolt-action rifle with a Mauser-inspired action renowned for its strength.  Though the rifle was historically sneered at in the US due to extensive anti-Japanese propaganda and the lower quality of later-production guns, the weapon is, by most metrics, a finely produced and effective shoulder arm.  This particular Arisaka is an early-war example with anti-aircraft sights and an intact 6-petal chrysanthemum seal.",
```

File: data/json/items/gun/77mm_jap.json [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        "weight": 1
+      },
+      {
+        "id": "type99ld",
```
Comment (TheShadowFerret) [inline data/json/items/gun/77mm_jap.json:18]: Does the ld here mean "last-ditch," like in the description 🤔

File: data/json/items/gunmod/sights.json [hunk 4]
```diff
@@ -317,5 +317,24 @@
...
     "flags": [ "DISABLE_SIGHTS", "CONSUMABLE", "ZOOM" ]
+  },
+  {
+    "id": "type99_scope",
+    "type": "GUNMOD",
+    "copy-from": "rifle_scope",
+    "name": { "str": "Nagoya 99 gun scope" },
+    "//": "Kokura also made and issued 2.5x rifle scopes for the Type 99, but those are vanishingly rare when compared against Nagoya-built, non-adjustable 4x scopes. That being said, Nagoya also built a small number of 2.5x scopes and dabbled with an adjustable 4x close to the end of production, but those are also comparatively rare enough to be hand-waved away.",
+    "description": "A basic and antique 4x rifle optic: essentially a non-adjustable telescopic sight built by the Nagoya Arsenal during the Second World War and issued for mounting on Type 99 sniper rifles.",
+    "range_multiplier": 3,
```
Comment (TheShadowFerret) [inline data/json/items/gunmod/sights.json:328]: Shouldn't this have a different range multipler than the x2.5 scope 🤔

File: data/json/items/magazine/65mm_jap.json [hunk 5]
```diff
@@ -0,0 +1,20 @@
+[
+  {
+    "id": "clip_65mm_arisaka",
+    "looks_like": "stanag30",
+    "type": "MAGAZINE",
+    "name": { "str": "6.5x50mm 5-round stripper clip" },
+    "description": "A tiny piece of grooved metal designed to allow a shooter to quickly load 5 6.5x50mm rounds into a compatible firearm.",
```
Comment (TheShadowFerret) [inline data/json/items/magazine/65mm_jap.json:7]:
```suggestion
    "description": "A tiny piece of grooved metal designed to allow a shooter to quickly load five 6.5x50mm rounds into a compatible firearm.",
```

File: data/json/items/magazine/77mm_jap.json [hunk 6]
```diff
@@ -0,0 +1,20 @@
+[
+  {
+    "id": "clip_77mm_arisaka",
+    "looks_like": "stanag30",
+    "type": "MAGAZINE",
+    "name": { "str": "7.7x58mm 5-round stripper clip" },
+    "description": "A tiny piece of grooved metal designed to allow a shooter to quickly load 5 7.7x58mm rounds into a compatible firearm.",
```
Comment (TheShadowFerret) [inline data/json/items/magazine/77mm_jap.json:7]:
```suggestion
    "description": "A tiny piece of grooved metal designed to allow a shooter to quickly load five 7.7x58mm rounds into a compatible firearm.",
```

File: data/json/items/melee/swords_and_blades.json [hunk 7]
```diff
@@ -3696,6 +3696,34 @@
     "flags": [ "DURABLE_MELEE", "SHEATH_SWORD" ],
     "weapon_category": [ "KNIVES" ]
   },
+  {
+    "id": "arisaka_bayonet",
+    "type": "TOOL",
+    "category": "weapons",
+    "name": { "str": "Type 30 bayonet" },
+    "description": "An essential tool within any 20th-century Japanese soldier’s toolbox, the type 30 is a massive, single-edged sword-style bayonet intended for use with Arisaka rifles and even light machineguns.  Possessing a J-shaped quillon guard for hooking an opponent’s weapon during combat, the bayonet, when dismounted, acts like a particularly sizable machete.",
```
Comment (TheShadowFerret) [inline data/json/items/melee/swords_and_blades.json:3704]:
```suggestion
    "description": "An essential tool within any 20th-century Japanese soldier’s toolbox, the Type 30 is a massive, single-edged sword-style bayonet intended for use with Arisaka rifles and even light machineguns.  Possessing a J-shaped quillon guard for hooking an opponent’s weapon during combat, the bayonet, when dismounted, acts like a particularly sizable machete.",
```

File: data/json/items/gun/77mm_jap.json [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+      [ "mechanism", 4 ],
+      [ "sights mount", 1 ],
+      [ "sling", 1 ],
+      [ "grip mount", 1 ],
+      [ "rail mount", 1 ],
+      [ "stock mount", 1 ],
+      [ "stock accessory", 2 ],
+      [ "underbarrel", 1 ]
+    ],
+    "//": "Having a single underbarrel slot precludes the mounting of both the monopod and the bayonet at the same time, but it's the only way to keep players from just stripping off the two and whacking on two underbarrel launchers or something.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/77mm_jap.json:51]: Given the monopod is only usable on this rifle & folds up anyways, why not make it an integrated gunmod?

~~Puts on another monopod~~

File: data/json/items/gunmod/underbarrel.json [hunk 9]
```diff
@@ -677,5 +677,50 @@
...
+    },
+    "min_skills": [ [ "weapon", 2 ] ],
+    "melee_damage": { "bash": 10 }
+  },
+  {
+    "id": "arisaka_monopod_deployed",
+    "type": "GUNMOD",
+    "name": { "str": "wire monopod, deployed" },
+    "description": "A folding wire monopod designed to attach to a Type 99 Arisaka’s front barrel band, intended to give soldiers a stabler platform from which to shoot while prone.  Less effective than an actual bipod and ungainly long, it was largely cut from active issue relatively soon into the Second World War.  This one is currently locked in the unfolded position and will provide a measure of stability while firing.",
+    "copy-from": "arisaka_monopod",
```
Comment (TheShadowFerret) [inline data/json/items/gunmod/underbarrel.json:714]: The integral_volume of this is still 0ml when deployed, I think it should take up full volume if it's out.

File: data/json/items/gun/77mm_jap.json (hunk 3 above)
Comment (DoctorBoomstick) [reply data/json/items/gun/77mm_jap.json:18]: Sorry for taking a while to get to these, I was away for a stretch. “ld” does stand for last-ditch. Felt like the id would have been too huge if I didn’t abbreviate it.

File: data/json/items/gunmod/sights.json (hunk 4 above)
Comment (DoctorBoomstick) [reply data/json/items/gunmod/sights.json:328]: According to my reading, the effective range modifiers were around a factor of three for both scopes: 1500 metres for a Type 97 with a telescopic sight as opposed to 500 metres with a normal 38, and 1500 metres for a Type 99 with a gun sight versus 657 metres for regular iron sights. Considering the numbers, I suppose the 4x could be nocked down to a 2-times modifier, or we could just calculate the distance in tiles when factoring range compression and work from there. As for the other two statistics, I didn’t particularly understand how the aim speed modifier affected the weapon from a game mechanics standpoint, and I couldn’t tell if I should tweak the field of view numbers.

File: data/json/items/gun/77mm_jap.json (hunk 8 above)
Comment (DoctorBoomstick) [reply data/json/items/gun/77mm_jap.json:51]: I didn’t make it a built-in or default gun mod as, according to what I’ve seen and read, the monopod is removable and is often missing from the rifles, as soldiers found that it wasn't particularly useful, and they were the first to get stripped from issue when material shortages knocked at the door.

**Review by kevingranade — CHANGES_REQUESTED** [review]
Sorry, unless you have some sources stating that working models are a lot more common than what is implied by "many thousands", followed by years of likely neglect or outright defacement, these are definitely not common enough to have in the game, especially not in the numbers you have here.

**Comment by DoctorBoomstick** [comment]
@kevingranade, May I make the suggestion that we could make do with some form of metric system to determine which guns should make it into the game and which shouldn’t, based on examining second-hand gun markets and auction sights? While it wouldn’t tell us the pure amounts of any given arm in circulation, it could at least indicate how roughly common they are by volume sold. For example, if we take a look at Gun Broker for SMLE rifles, there are 211 hits, while 446 hits register for No. 4 Enfields. M1918 BARs come out to 40 hits, while in the Arisaka camp we’ve got 232 hits for Type 99s, 128 for Type 38s, and 86 for Type 30s. So, they’re out there, and in large enough numbers for a comfortable amount of complete rifles to be traded around. When you also take into consideration that second-hand markets don’t show how many rifles are privately owned or in circulation, the number would come out to a fair bit more. If I removed the common spawns and tweaked down the rare spawns, would you consider this acceptable?

**Comment by GuardianDll** [comment]
After some investigation, using metric, proposed by Tenmillimaster, as "has 100 hits or more on a gun brocker for a family of firearms", and watching at selling history, it seems there is indeed more than 100 hits overall (it shows overall, but some should be excluded because it's either some gun parts, that happened to be in the rifle category, or sporterized, which may not count in this calculations)
image

**Comment by DoctorBoomstick** [comment]
> After some investigation, using metric, proposed by Tenmillimaster, as "has 100 hits or more on a gun brocker for a family of firearms", and watching at selling history, it seems there is indeed more than 100 hits overall (it shows overall, but some should be excluded because it's either some gun parts, that happened to be in the rifle category, or sporterized, which may not count in this calculations) image


So, which rifles do you think merit staying and which should be booted? I would tentatively prepose that the rarer guns could be moved to the rifle_obscure item group, rather than outright removed, though idk what manner of commonality rifles should possess before they’re placed in there.

**Comment by kevingranade** [comment]
Looks like the 7.7 scrapes by over the threshold, and those still belong in the rifle_obscure group.

**Pull request merged**